### ISSUE TITLE
Incorrect syntax highlighting even after 1.32.1 on params, imports and variable declarations

### ISSUE THREAD

**Issue by CorentinAndre** [atom/language-javascript#615]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

With the new Tree Sitter Parsers functionnality, params highlighting is broken as well as const declaration and imports, even on Atom 1.32.1
Here is a picture showing what's wrong.
![capture d ecran 2018-10-31 a 11 48 18](https://user-images.githubusercontent.com/11957741/47783727-c71ce480-dd03-11e8-9486-cf72de8d9fe6.png)

### Steps to Reproduce

1. Enable Tree Sitter Parsers functionnality

**Expected behavior:** Correct Highlighting
![capture d ecran 2018-10-31 a 12 14 36](https://user-images.githubusercontent.com/11957741/47784669-8f636c00-dd06-11e8-9d7a-b2c3ae2812b7.png)

**Actual behavior:** See picture in description

**Reproduces how often:** 100%

### Versions

Atom    : 1.32.1
Electron: 2.0.9
Chrome  : 61.0.3163.100
Node    : 8.9.3

MacOS Mojave 10.14

**Comment by benmumma** [comment]
I'm having the same issue on a Windows

**Comment by Buzut** [comment]
Same problem here. It's especially annoying that function documentation is not hilighted at all. In the following exemple, all ƒn's doc would stay white. 

```javascript
/**
 * Set an event listener on an element
 * @param { HTMLElement } el
 * @param { String } action The event to listen to
 * @param { Function } fn Function to execute when event triggers
 */
export function on(el, action, fn) {
    el.addEventListener(action, fn);
}
```

**Comment by Arcanemagus** [comment]
@Buzut You will want to follow https://github.com/atom/language-javascript/issues/600 for when that is fixed.

**Comment by maxbrunsfeld** [comment]
👋 This is the new expected behavior; some colors have changed.

The old behavior was very inconsistent; it often led to a given name being colored differently in different places, and we've tried to reduce that as much as possible now that we have a more accurate way of understanding code.

We don't assign distinct colors to variables declared with `const` and `import` because we don't (and never did) have a way of tracking these variables reliably throughout the file. So if we colored them, say, red, inside the `const` or `import` declarations, they would still be colored white in other places, creating an inconsistency. We believe this inconsistency overall does much more harm than good for the readability of code.
